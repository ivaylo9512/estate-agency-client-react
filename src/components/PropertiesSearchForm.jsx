import useInput from "hooks/useInput"
import useMinMaxInput from "hooks/useMinMaxInput";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faBed } from '@fortawesome/free-solid-svg-icons'
import { useDispatch, useSelector } from "react-redux";
import { getProperties, resetPropertiesState } from "app/slices/propertiesPaginationSlice";
import { useState, useEffect } from "react";

const PropertiesSearchForm = () => {
    const [values, {locationInput, minPriceInput, maxPriceInput}] = createInputs();
    const [bedroom, setBedrooms] = useState();
    const dispatch = useDispatch();
    const query = useSelector(state => state.propertiesPagination.query);

    useEffect(() => {
        return () => dispatch(resetPropertiesState()) 
    },[])

    const submit = (e) => {
        e.preventDefault();

        if(values.location){
            const queryValues = {...query, ...values, pages: 1, bedrooms: e.target.bedrooms.value};
            dispatch(resetPropertiesState())
            dispatch(getProperties(queryValues))
        }
    }

    return(
        <form onSubmit={submit}>
            {locationInput}
            {minPriceInput}
            {maxPriceInput}
            <label htmlFor='bedroom'><FontAwesomeIcon icon={faBed}/></label>
            <select onChange={(e) => setBedrooms(e.target.value)} name='bedrooms'>
                <option value='0'>any</option>
                <option>1</option>
                <option>2</option>
                <option>3</option>
                <option>4</option>
                <option>5</option>
                <option>6</option>
                <option>7</option>
            </select>
            <button />
        </form>
    )
}
export default PropertiesSearchForm

const createInputs = () => {
    const [location, locationInput] = useInput({
        name: 'location',
        placeholder: 'location',
        initialValue: ''
    });

    const [minPrice, maxPrice, minPriceInput, maxPriceInput] = useMinMaxInput({
        minName: 'fromPrice',
        maxName: 'toPrice',
        minValue: 0,
        maxValue: 5000 * 1000,
        minInitial: 0,
        maxInitial: 5000 * 1000
    });

    return [{location, minPrice, maxPrice}, {locationInput, minPriceInput, maxPriceInput}]
} 