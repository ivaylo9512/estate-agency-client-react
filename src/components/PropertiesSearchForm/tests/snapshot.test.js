import PropertiesSearchForm from '../PropertiesSearchForm';
import * as Redux from 'react-redux';
import { shallow } from 'enzyme';

describe('PropertiesSearchForm snapshot tests', () => {
    beforeAll(() => {
        jest.spyOn(Redux, 'useDispatch').mockReturnValue(jest.fn());
        jest.spyOn(Redux, 'useSelector').mockReturnValue(jest.fn());
    })

    const createWrapper = () => shallow(
        <PropertiesSearchForm />
    )

    it('should match snapshot', () => {
        const wrapper = createWrapper(true);

        expect(wrapper).toMatchSnapshot();
    })
})