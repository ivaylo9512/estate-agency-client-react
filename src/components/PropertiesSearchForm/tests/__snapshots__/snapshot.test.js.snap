// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertiesSearchForm snapshot tests should match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="location"
    onChange={[Function]}
    placeholder="location"
    type=""
    value=""
  />
  <input
    min={0}
    name="fromPrice"
    onChange={[Function]}
    type="number"
    value={0}
  />
  <input
    max={5000000}
    name="toPrice"
    onChange={[Function]}
    type="number"
    value={5000000}
  />
  <label
    htmlFor="bedroom"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            640,
            512,
            Array [],
            "f236",
            "M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z",
          ],
          "iconName": "bed",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </label>
  <select
    name="bedrooms"
    onChange={[Function]}
  >
    <option
      value="0"
    >
      any
    </option>
    <option>
      1
    </option>
    <option>
      2
    </option>
    <option>
      3
    </option>
    <option>
      4
    </option>
    <option>
      5
    </option>
    <option>
      6
    </option>
    <option>
      7
    </option>
  </select>
  <button />
</form>
`;
